### PR TITLE
feat(i18n): add translation resources

### PR DESCRIPTION
**Pull request by ocBruno**
I plan on looking for bilingual volunteers to help properply translate these documents and correct the mistakes if anyone has any suggestions please let me know!

**Changes proposed in this pull request:**
- Add 8 other most used languages on the internet to static i18n translation resources with initial
rough translations from google tranlsate.

### CODE REVIEW

**Comment by fox1t** [comment]
First of all, thanks for this PR! It is really appreciated :)

Secondly I think we need better folder and file structure.  We can use "app sections" as top level folder and the inside it create language sub-folders. What do you think about that?

**Comment by eacdev** [comment]
Hi! I can do French, it is also a widely spoken language in Africa :)

**Comment by ocBruno** [comment]
> As @fox1t mentioned, we should organize the translations by app sections.
> 
> For example instead of the `app-sections` folder there would be a `patients` folder and `appointments` folder.

Back from vacation so i'm still a little rusty lol! Not sure if this structure is what you were thinking and I was in doubt of what an ideal filename for the text to be translated would be so I left translation.json for now. If anything send a request!

**Comment by jackcmeyer** [comment]
This looks like the structure we would want. However, it isn’t working. If you check out the preview, you can see it’s not rendering the correct text

**Comment by ocBruno** [comment]
> This looks like the structure we would want. However, it isn’t working. If you check out the preview, you can see it’s not rendering the correct text

Hey @jackcmeyer, I wanted to make sure the file structure was correct before moving on to the next steps so thanks for confirming that for me.Also I was absent for some time for vacation so I'm still a little rusty. I also saw that @tehKapa assigned you to this so I wasn't sure if I was supposed to continue contributing but I thought there's nothing to lose lol! 

If there is any help you guys need in design, ux or if there are some issues for begginers thar are better suited for me send me a message!

Until then I'll try to get everything working over here for the static translations before committing and pushing because I saw there was some automatic deployments I wasn't aware of if not I would've held back those pushes. If I make some progress I'll ask before pushing my next commits if necessary!

Thanks for the feedback and if I anything appears don't hesitate to message me!

**Comment by ocBruno** [comment]
> This looks like the structure we would want. However, it isn’t working. If you check out the preview, you can see it’s not rendering the correct text

I have resolved the issue that was preventing the build and am back to tackle this issue, if anyone sees anything wrong with my commits or something that could be improved send me heads up!

**Comment by ocBruno** [comment]
> As @fox1t mentioned, we should organize the translations by app sections.
> 
> For example instead of the `app-sections` folder there would be a `patients` folder and `appointments` folder.

Hey @jackcmeyer,
I thought I had wrapped this one up but realized I forgot to separate into folders, I wasn't sure if we were supposed to use json in the separate files or a different method but I decided the typescript approach.
 If anything leave me a change request!

**Comment by ocBruno** [comment]
Hey @jackcmeyer,
just letting you know despite the I updated this branch and refactored the translations into separate folders as you suggested.

**Comment by ocBruno** [comment]
Hello @jackcmeyer @fox1t @irvelervel and @tehKapa,
Despite a couple difficulties, I updated this branch and refactored the translations into their own separate folders as requested and fixed a couple test and lint issues along the way. 
If anyone has any ideas on the next steps we should take to improve the integration of i18next and internationalization leave me a comment! 

An idea I have is for the static resources we should gradually add the necessary translations with the help of other bilingual people. We can place a good first issue to leave pinned on the issues page which attracts new contributors worldwide with a simple first issue. 

Until there are other necessary changes I'm going to move on to other issues but if there's anything missing to close this pull request give me a shout!

**Comment by fox1t** [comment]
> Hello @jackcmeyer @fox1t @irvelervel and @tehKapa,
> Despite a couple difficulties, I updated this branch and refactored the translations into their own separate folders as requested and fixed a couple test and lint issues along the way.
> If anyone has any ideas on the next steps we should take to improve the integration of i18next and internationalization leave me a comment!


> An idea I have is for the static resources we should gradually add the necessary translations with the help of other bilingual people. We can place a good first issue to leave pinned on the issues page which attracts new contributors worldwide with a simple first issue.
> 

Nice idea. I completely agree that we need an issue to track all languages we support, where we can update a checklist (pinned). Then we will open an issue for a single language.

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -15,31 +15,27 @@ const Navbar = () => {
...
       search={{
-        onClickButton: () => {
-          // no oop
-        },
-        onChangeInput: () => {
-          // no oop
-        },
-        placeholderText: t('actions.search'),
-        buttonText: t('actions.search'),
+        buttonText: 'actions.search',
```
Comment (fox1t) [inline src/components/Navbar.tsx]: is `t(` missing here?

**Comment by ocBruno** [comment]
> Looks like there is some missing translations (check out the preview and you can see the ones that are missing)

Hey @jackcmeyer,
I'm going to try to keep the english and portuguese translations up to date and I can help search for some bilingual contributors to help us out, create some good first issues or something along those lines.

**Comment by tehKapa** [comment]
@ocBruno any news?

**Comment by ocBruno** [comment]
> @ocBruno any news?
@tehKapa 
I had updated english and portuguese to all necessary translations, not sure what else, if anyone has any other requests drop them here .

**Comment by fox1t** [comment]
@ocBruno Any news about this PR? I think this is a needed feature and you done a good job.

**Comment by ocBruno** [comment]
> @ocBruno Any news about this PR? I think this is a needed feature and you done a good job.

Hey @fox1t, 

I started working two weeks ago at a new company and have less free time, if someone wants to keep working on this feel free but for now I will be working on one issue at a time and I am trying to wrap up the input-validation on components library before returning to this one.

Last time I checked this I was unsure exactly what else was needed to finish. I think we should create an issue for separate languages for new contributors to do some translation for the static website data.

**Comment by fox1t** [comment]
@ocBruno there is only one thing left to do: resolve the conflicts. After it, we can land it on master! :)

**Comment by jackcmeyer** [comment]
Moving this to #1892

**Pull request closed without merging**